compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -722,6 +722,7 @@ public void visit(BLangLimitClause limitClause) {
     public void visit(BLangWhile whileNode) {
         lookupNode(whileNode.expr);
         lookupNode(whileNode.body);
+        lookupNode(whileNode.onFailClause);
```

Comment:
We have to do the similar fix for the ForEach and Lock statement as well. @malinthar can you please do that?

Reply:
Will do